src/Common/MemoryTrackerSwitcher.h

```diff
@@ -31,8 +32,10 @@ struct MemoryTrackerSwitcher
         CurrentThread::flushUntrackedMemory();
         auto * thread_tracker = CurrentThread::getMemoryTracker();
 
-        current_thread->untracked_memory = prev_untracked_memory;
+        /// It is important to set untracked memory after the call of
+        /// 'setParent' because it may flush untracked memory to the wrong parent.
         thread_tracker->setParent(prev_memory_tracker_parent);
+        current_thread->untracked_memory = prev_untracked_memory;
```

Comment:
Consider that we have the common chain of memory trackers: `Thread <- Process <- Global` and we created `MemoryTrackerSwitcher` with `total_memory_tracker`. Here in destructor we have the chain `Thread <- Global`. If we set untracked memory before `setParent` then according to [this](https://github.com/ClickHouse/ClickHouse/blob/d2d8618d271353b42aa3a6eeb2652a47204ba662/src/Common/MemoryTracker.cpp#L595-L598) code it will be flush to the current parent which is `Global` and will not be accounted in `Process` memory tracker.